Objective-C/CBLDocument.mm

```diff
@@ -210,9 +223,9 @@ - (void) replaceC4Doc: (CBLC4Document*)c4doc {
 
 #pragma mark - Fleece Encoding
 
-- (FLSliceResult) encodeWithRevFlags: (C4RevisionFlags*)outRevFlags error:(NSError**)outError {
+- (FLSliceResult) encodeWithRevFlags: (C4RevisionFlags*)outRevFlags shared: (BOOL)shared error:(NSError**)outError {
```

Comment:
If I understand it correctly, the reason not to use the shared fleece encoder because we don't want to do shared key encoding. Renaming `shared:` label to `useSharedKeys` or `useSharedEncoder` is better.

Reply:
Yea, also shared encoder needs the c4db.